### ISSUE TITLE
0.9.31 Mobile Release

### ISSUE THREAD

**Issue by chadyj** [status-im/status-react#6681]
This is an issue to list and discuss remaining issues for the 0.9.31 release.

Remaining issues can be viewed with the release tag.
https://github.com/status-im/status-react/labels/release

We will use this issue to manage the release. (see #6311 for an example)

Release notes: https://notes.status.im/mobile-release-notes
Changelog: https://github.com/status-im/status-react/compare/release/0.9.30...release/0.9.31

**Comment by chadyj** [comment]
0.9.31 will be cut this Wednesday.

Check the [commit log for what's new](https://github.com/status-im/status-react/compare/release/0.9.30...release/0.9.31).

**Comment by lukaszfryc** [comment]
I am starting nightly testing now. Hopefully I will not find any blockers and https://github.com/status-im/status-react/pull/6673 will be merged soon. Then we can cut the release branch.

**Comment by rachelhamlin** [comment]
@lukaszfryc I was slow to add the `release` label to this issue—if we can include it that would be great: https://github.com/status-im/status-react/issues/6693 (needs to be merged @jeluard )

**Comment by lukaszfryc** [comment]
Sure we can. The release branch is not created yet.

**Comment by lukaszfryc** [comment]
So far we have not found any issues. But, the nightly testing will probably continue till the end of the day.

**Comment by lukaszfryc** [comment]
@chadyj @mandrigin we tested 80% of `e8d3e3` nightly and not found any issues. I think we can cut the release tomorrow and continue with final release tests. 

We can cut from top of develop. Apart from https://github.com/status-im/status-react/commit/1104becfbab11506b98028ce81ad6ff869381f51 there is no other big change. @cammellos is additional testing needed for it? We already finished nightly testing for another version (`e8d3e3`). I'm not sure if it's safe to include 1104becfbab11506b98028ce81ad6ff869381f51 in release.

**Comment by lukaszfryc** [comment]
Also, I went through all issues reported after 0.9.30 and have not found any new bugs that would block the release.

**Comment by cammellos** [comment]
i am looking at the offline / online issue, if fixed it would be good to
include, pairing should be safe to include as disabled, i can check the rc
to make sure it's the case, but can be excluded if we want to be cautious

**Comment by lukaszfryc** [comment]
> i am looking at the offline / online issue, if fixed it would be good to
include, pairing should be safe to include as disabled, i can check the rc
to make sure it's the case, but can be excluded if we want to be cautious

Ok, let's include it.

@mandrigin or @cammellos could you cut the release/0.9.31 from the current develop (`89e0ea0bd2211a2f0462488ebf1166d592cd064d`)? We want to start the final release testing asap.

cc @chadyj

**Comment by chadyj** [comment]
Heya @mandrigin @lukaszfryc what do you think about a new doc on how to cut a release for dummies?

- How to cut the branch
- What buttons to push
- etc

**Comment by cammellos** [comment]
here's the branch https://github.com/status-im/status-react/tree/release/0.9.31

<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>a free (libre) open source, mobile OS for Ethereum - status-im/status-react</div></blockquote>

**Comment by chadyj** [comment]
Thanks @cammellos 

Release notes up at https://notes.status.im/mobile-release-notes

cc @j-zerah

**Comment by lukaszfryc** [comment]
>i am looking at the offline / online issue, if fixed it would be good to
include, pairing should be safe to include as disabled, i can check the rc
to make sure it's the case, but can be excluded if we want to be cautious


@cammellos here are release builds. You can double check them:

https://ci.status.im/job/status-react/job/combined/job/mobile-android/7385/artifact/StatusIm-181115-082235-1104be-e2e.apk
https://ci.status.im/job/status-react/job/combined/job/mobile-ios/3710/artifact/StatusIm-181115-082235-1104be-release.ipa

<blockquote></blockquote>
<blockquote></blockquote>

**Comment by lukaszfryc** [comment]
OK, testing is done. No new issues found. Here is the latest [testrail testrun](https://ethstatus.testrail.net/index.php?/runs/view/1964&group_by=cases:section_id&group_order=asc) (critical section was tested on the final build).

I think we can publish the app once https://github.com/status-im/status-react/pull/6742 is merged

cc @chadyj @mandrigin @asemiankevich

**Comment by lukaszfryc** [comment]
> > i am looking at the offline / online issue, if fixed it would be good to
> > include, pairing should be safe to include as disabled, i can check the rc
> > to make sure it's the case, but can be excluded if we want to be cautious
> 
> cammellos here are release builds. You can double check them:
> 
> https://ci.status.im/job/status-react/job/combined/job/mobile-android/7385/artifact/StatusIm-181115-082235-1104be-e2e.apk
> https://ci.status.im/job/status-react/job/combined/job/mobile-ios/3710/artifact/StatusIm-181115-082235-1104be-release.ipa

@cammellos have you got a chance to look at it?

**Comment by cammellos** [comment]
@lukaszfryc sorry, not yet, I get a 404 for the android link, is there another link i can use?

**Comment by lukaszfryc** [comment]
@cammellos you can use https://ci.status.im/job/status-react/job/combined/job/mobile-android/7434/artifact/StatusIm-181115-130912-1104be-e2e.apk from [latest build](https://ci.status.im/blue/organizations/jenkins/status-react%2Frelease/detail/release%2F0.9.31/4/pipeline/44).

<blockquote></blockquote>

**Comment by chadyj** [comment]
@mandrigin can you help us out with a guide?

From memory we have to do:

- Cut branch (already done!)
- Bump version (eg [here we bumped 0.9.30](https://github.com/status-im/status-react/pull/6102/files))
- Run the [upload_release_ios](https://ci.status.im/job/status-react/job/upload_release_ios/) Jenkins job, and get approval. we can keep running to add new builds
- When iOS is good we run the [upload_release_android](https://ci.status.im/job/status-react/job/upload_release_android/)

Then the rest can be done in the App Store managers online.

**Comment by cammellos** [comment]
@lukaszfryc the one you sent me it's an e2e version, in that one group chats are enabled, we need one that is using `.env.prod` jenkins file

**Comment by cammellos** [comment]
should I kick the build?

**Comment by chadyj** [comment]
Can a @status-im/clojure dev please merge and cherry pick this PR? https://github.com/status-im/status-react/pull/6742

**Comment by cammellos** [comment]
@chadyj which PR?

**Comment by cammellos** [comment]
#6742 has been merged and cherry-picked

**Comment by chadyj** [comment]
Cheers!
Started https://ci.status.im/job/status-react/job/upload_release_ios/54/ so we can get a jump on approval.

<blockquote><img src="/static/60651f1e/favicon.ico" width="48" align="right"><div><strong><a href="https://ci.status.im/job/status-react/job/upload_release_ios/54/">status-react » upload_release_ios #54 [Jenkins]</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
> @lukaszfryc the one you sent me it's an e2e version, in that one group chats are enabled, we need one that is using `.env.prod` jenkins file

@cammellos we do not have it yet since jenkins build is broken. See https://github.com/status-im/status-react/issues/6743. Maybe you can help?

**Comment by chadyj** [comment]
~~It has been pretty quiet around here so I thought I'd see if I can fix the Jenkins build problem in https://github.com/status-im/status-react/pull/6751~~ Already done

~~Also the https://ci.status.im/job/status-react/job/upload_release_ios/54/  release job completed and 0.9.31 is in Testflight now for testing and approval.~~ https://github.com/status-im/status-react/pull/6750 needs to be merged first

**Comment by cammellos** [comment]
@lukaszfryc checked on my side, it's all good, all disabled

**Comment by chadyj** [comment]
*&^%$%^&* Jenkins failed again for the upload_release_ios job.  Running it again.
https://ci.status.im/job/status-react/job/upload_release_ios/56/

cc @jakubgs @mandrigin Multiple Jenkins failures seem to happen with every release waisting everyone's time.   eg https://github.com/status-im/status-react/issues/6743 and release job failures.   Also https://ci.status.im/job/status-react/job/upload_release_android/ hasn't successfully run in 2 months so I'm expecting more problems.

<blockquote><img src="/static/60651f1e/favicon.ico" width="48" align="right"><div><strong><a href="https://ci.status.im/job/status-react/job/upload_release_ios/56/">status-react » upload_release_ios #56 [Jenkins]</a></strong></div></blockquote>
<blockquote><img src="/static/60651f1e/favicon.ico" width="48" align="right"><div><strong><a href="https://ci.status.im/job/status-react/job/upload_release_android/">upload_release_android [status-react] [Jenkins]</a></strong></div></blockquote>

**Comment by chadyj** [comment]
Build 12473 has been submitted to Apple for approval.

**Comment by chadyj** [comment]
Happy Monday y'all! 🎉 
It's Monday and we don't have any new blockers.

Has the testing wrapped up?

**Comment by lukaszfryc** [comment]
@chadyj yes, testing is done.

**Comment by chadyj** [comment]
Perfect!

Build 12473 is ready to go in the Play Store and Testflight.

Release notes are ready https://notes.status.im/mobile-release-notes

@j-zerah When do you want to push the button?

cc @mandrigin @lukaszfryc @annadanchenko @rachelhamlin 

<blockquote><div><strong><a href="https://notes.status.im/mobile-release-notes">Mobile Release Notes - CodiMD</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@chadyj wait a sec please. There is an issue test team is investigating right now. Please join #test-team on Status.

**Comment by annadanchenko** [comment]
so, here is an issue discussed now in test-team: https://github.com/status-im/status-react/issues/6768
imo, it looks like a release blocker cc @chadyj @mandrigin

**Comment by annadanchenko** [comment]
it looks like test-team will need to retest upgrade again as wrong version from TestFlight was considered to be 0.9.30 release (we have 4 different options there).

 @lukaszfryc will you be able to check it using upgrade from 0.9.30 (11274)?

**Comment by lukaszfryc** [comment]
@annadanchenko yes, I will test it now.

**Comment by chadyj** [comment]
> as wrong version from TestFlight was considered to be 0.9.30 release (we have 4 different options there).

Let's start expiring builds so we only ever have 1 active build for each version. 
I expired the other 3 for 0.9.30.

**Comment by lukaszfryc** [comment]
I have not found other issues. I will retest upgrade scenarios after #6768 is fixed.

**Comment by rachelhamlin** [comment]
@chadyj @j-zerah we need to get the blog asset from Alex. I'll format the notes into a ghost post.

**Comment by chadyj** [comment]
0.9.31 is released.

For 0.9.32 let's aim to have release documentation so it is clear to everyone how to cut a release.

I started a doc here https://notes.status.im/mobile-release-guide

<blockquote><div><strong><a href="https://notes.status.im/mobile-release-guide">Mobile Release Guide - CodiMD</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
@chadyj why it was released without https://github.com/status-im/status-react/issues/6768? It had the release label and was mentioned here couple of times.

**Comment by chadyj** [comment]
cc @Blockchain-Islander
